### PR TITLE
RavenDB-21996 IDocumentQueryBase and IRavenQueryable documentation

### PR DESCRIPTION
**Pull request by Lwiel**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-21996

### Additional description

IDocumentQueryBase and IRavenQueryable documentation update

### Type of change

- [ ] Bug fix
- [ ] Regression bug fix
- [ ] Optimization
- [ ] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [x] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

File: src/Raven.Client/Documents/Session/AbstractDocumentQuery.Includes.cs [hunk 1]
```diff
@@ -26,7 +26,7 @@ public abstract partial class AbstractDocumentQuery<T, TSelf>
         /// <summary>
         ///   Includes the specified path in the query, loading the document specified in that path
         /// </summary>
-        /// <param name = "path">The path.</param>
+        /// <param name = "path">Path to included document.</param>
```
Comment (arekpalinski) [inline src/Raven.Client/Documents/Session/AbstractDocumentQuery.Includes.cs]: Can we reference all Include method to the following docs article:
https://ravendb.net/docs/article-page/6.0/Csharp/client-api/how-to/handle-document-relationships#includes ?

Comment (arekpalinski) [reply src/Raven.Client/Documents/Session/AbstractDocumentQuery.Includes.cs]: Also what is a bit missing here is some explanation what are includes. We mention its for loading "the document specified in that path" but it doesn't say anything. What is more important is that it gives ability to load _related_ / _referenced_ documents. That's the point. Includes is a very important feature - please try to make its documentation more extensive (here and in other places as well).

File: src/Raven.Client/Documents/Session/IAbstractDocumentQuery.cs [hunk 2]
```diff
@@ -17,61 +17,66 @@ namespace Raven.Client.Documents.Session
...
         /// </summary>
         void RandomOrdering(string seed);
 
 #if FEATURE_CUSTOM_SORTING
         /// <summary>
-        /// Sort using custom sorter on the server
+        ///     Sort query results using server-side custom sorter.
+        ///     Requires custom sorting feature to be enabled.
         /// </summary>
+        /// <param name="descending">Changes order to descending.</param>
```
Comment (arekpalinski) [inline src/Raven.Client/Documents/Session/IAbstractDocumentQuery.cs:69]: Can we reference to https://ravendb.net/docs/article-page/6.0/csharp/client-api/session/querying/sort-query-results#custom-sorters ?

File: src/Raven.Client/Documents/Session/IDocumentQueryBase.cs [hunk 3]
```diff
@@ -434,39 +434,35 @@ public interface IGroupByDocumentQueryBase<T, TSelf> where TSelf : IDocumentQuer
     }
 
     /// <summary>
-    ///     A query against a Raven index
+    ///     Interface for lower level API access.
```
Comment (arekpalinski) [inline src/Raven.Client/Documents/Session/IDocumentQueryBase.cs]: Interface for lower level API access. -> Interface providing low-level querying capabilities

File: src/Raven.Client/Documents/Session/IDocumentQueryBase.cs [hunk 4]
```diff
@@ -482,265 +478,331 @@ public interface IDocumentQueryBase<T, TSelf> : IPagingDocumentQueryBase<T, TSel
         int Count(Func<T, bool> predicate);
 
         /// <summary>
-        ///     Apply distinct operation to this query
+        ///     Removes duplicates from query results.
```
Comment (arekpalinski) [inline src/Raven.Client/Documents/Session/IDocumentQueryBase.cs:481]: We have a dedicated article about Distinct:
https://ravendb.net/docs/article-page/6.0/csharp/indexes/querying/distinct
Why not to reference it?

File: src/Raven.Client/Documents/Session/AbstractDocumentQuery.Includes.cs (hunk 1 above)
Comment (Lwiel) [reply src/Raven.Client/Documents/Session/AbstractDocumentQuery.Includes.cs]: Changed

File: src/Raven.Client/Documents/Session/IAbstractDocumentQuery.cs (hunk 2 above)
Comment (Lwiel) [reply src/Raven.Client/Documents/Session/IAbstractDocumentQuery.cs:69]: Added

File: src/Raven.Client/Documents/Session/IDocumentQueryBase.cs (hunk 3 above)
Comment (Lwiel) [reply src/Raven.Client/Documents/Session/IDocumentQueryBase.cs]: Changed

File: src/Raven.Client/Documents/Session/IDocumentQueryBase.cs (hunk 4 above)
Comment (Lwiel) [reply src/Raven.Client/Documents/Session/IDocumentQueryBase.cs:481]: Added

**Review by arekpalinski — APPROVED** [review]

**Pull request merged**